app/web/features/auth/notifications/EditNotificationSettingsPage.tsx

```diff
@@ -0,0 +1,145 @@
+import Snackbar from "components/Snackbar";
+import { AUTH } from "i18n/namespaces";
+import { useTranslation } from "next-i18next";
+import { useEffect, useState } from "react";
+import makeStyles from "utils/makeStyles";
+
+import NotificationSettingsListItem from "./NotificationSettingsListItem";
+import useNotificationSettings from "./useNotificationSettings";
+
+export type NotificationType =
+  | "account_security"
+  | "account_settings"
+  | "chat"
+  | "event"
+  | "reference"
+  | "friend_request"
+  | "host_request";
+
+export interface GroupAction {
+  action: string;
+  description: string;
+  email: boolean;
+  push: boolean;
+  topic: string;
+  userEditable: boolean;
+}
+
+interface GroupsByType {
+  [key: string]: GroupAction[];
+}
+
+const useStyles = makeStyles((theme) => ({
+  descriptionText: {
+    fontSize: theme.spacing(1.8),
+    color: theme.palette.text.secondary,
+  },
+  list: {
+    border: `1px solid ${theme.palette.divider}`,
+    marginTop: theme.spacing(1),
+    display: "flex",
+    flexDirection: "column",
+    padding: `0 ${theme.spacing(1)}`,
+  },
+  loading: {
+    position: "absolute",
+  },
+  notificationSettingsContainer: {
+    display: "flex",
+    flexDirection: "column",
+    padding: theme.spacing(4),
+    margin: "0 auto",
+    width: "100%",
+    [theme.breakpoints.up("md")]: {
+      width: "50%",
+    },
+  },
+  notificationDescription: {
+    margin: theme.spacing(1, 0),
+    paddingBottom: theme.spacing(3),
+  },
+}));
+
+export default function EditNotificationSettingsPage() {
+  const classes = useStyles();
+  const { t } = useTranslation(AUTH, {
+    keyPrefix: "notification_settings.edit_preferences",
+  });
+  const { data, isLoading, isError } = useNotificationSettings();
+  const [groups, setGroups] = useState<GroupsByType>({});
+  const [areGroupsLoading, setAreGroupsLoading] = useState<boolean>(true);
+
+  useEffect(() => {
+    try {
+      if (!data) {
+        return;
+      }
+
+      const computedGroups = data?.groupsList.reduce<GroupsByType>(
+        (acc, group) => {
+          group.topicsList.forEach((topic) => {
+            if (topic && topic.itemsList) {
+              topic.itemsList.forEach((subTopic) => {
+                const key =
+                  group.heading === "Account Security"
+                    ? "account_security"
+                    : group.heading === "Account Settings"
+                    ? "account_settings"
+                    : topic.topic;
+
+                if (!acc[key]) {
+                  acc[key] = [];
+                }
+
+                if (subTopic.userEditable) {
+                  acc[key].push({ ...subTopic, topic: topic.topic });
+                }
+              });
+            }
+          });
+
+          return acc;
+        },
+        {}
+      );
+
+      setGroups(computedGroups);
+    } catch (error) {
+      console.error("Error fetching notification settings data:", error);
```

Comment:
Should you re-raise it here?

Reply:
What specifically are you thinking I should re-raise?

Reply:
Right now you are catching and suppressing any error that occurs in this block, right? If I understand correctly you are just ignoring the error and the user will see no groups?

When I do backend stuff we try to not suppress errors: only catch and handle the ones we can recover from, and let others bubble up to a global exception handler.

I guess on the frontend there's an error boundary. Of course it will look bad for the user if they get that, but it won't silently fail.

I'm not sure if this is right for the frontend, hence the qeustion :)

Reply:
Ah yes you're right I should generally def do that. On further look of this specific one though, I removed the try/catch. This is because the api call error handling is already being handled in the hook in this case, so the only thing this would be catching is if my parsing of the groups went wrong, which I would already know at this point.